client/my-sites/email/email-management/email-home.tsx

```diff
@@ -0,0 +1,207 @@
+import EmailNoDomain from 'calypso/my-sites/email/email-management/home/email-no-domain';
+import EmailPlan from 'calypso/my-sites/email/email-management/home/email-plan';
+import EmailProvidersComparisonStacked from 'calypso/my-sites/email/email-providers-stacked-comparison';
+import { emailManagementTitanSetUpMailbox } from 'calypso/my-sites/email/paths';
+import SidebarNavigation from 'calypso/my-sites/sidebar-navigation';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import hasLoadedSites from 'calypso/state/selectors/has-loaded-sites';
+import { createSiteDomainObject } from 'calypso/state/sites/domains/assembler';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { TranslateResult } from 'i18n-calypso';
+import type { ReactElement } from 'react';
+
+import './style.scss';
+
+interface ContentWithHeaderProps {
+	children: ReactElement;
+}
+
+const ContentWithHeader = ( { children }: ContentWithHeaderProps ): ReactElement => {
+	const translate = useTranslate();
+	return (
+		<Main wideLayout>
+			<DocumentHead title={ toTitleCase( translate( 'Emails', { textOnly: true } ) ) } />
+
+			<SidebarNavigation />
+
+			<EmailHeader />
+
+			{ children }
+		</Main>
+	);
+};
+
+const NoAccess = (): ReactElement => {
+	const translate = useTranslate();
+	return (
+		<ContentWithHeader>
+			<>
+				<EmptyContent
+					title={ translate( 'You are not authorized to view this page' ) }
+					illustration={ '/calypso/images/illustrations/illustration-404.svg' }
+				/>
+			</>
+		</ContentWithHeader>
+	);
+};
+
+const LoadingPlaceholder = (): ReactElement => {
+	return (
+		<ContentWithHeader>
+			<>
+				<SectionHeader className="email-home__section-placeholder is-placeholder" />
+				<Card className="email-home__content-placeholder is-placeholder" />
+			</>
+		</ContentWithHeader>
+	);
+};
+
+interface EmailManagementHomeProps {
+	emailListInactiveHeader: ReactElement;
+	sectionHeaderLabel: TranslateResult;
+	selectedDomainName: string;
+	showActiveDomainList?: boolean;
+	source: string;
+}
+
+const EmailHome = ( props: EmailManagementHomeProps ): ReactElement => {
+	const {
+		emailListInactiveHeader,
+		showActiveDomainList = true,
+		selectedDomainName,
+		sectionHeaderLabel,
+		source,
+	} = props;
+
+	const selectedSite = useSelector( getSelectedSite );
+
+	const canManageSite = useSelector( ( state ) => {
+		if ( ! selectedSite ) {
+			return;
+		}
+		return canCurrentUser( state, selectedSite.ID, 'manage_options' );
+	} );
+	const currentRoute = useSelector( ( state ) => getCurrentRoute( state ) );
+	const hasSitesLoaded = useSelector( ( state ) => hasLoadedSites( state ) );
+
+	const { data, isLoading: isSiteDomainLoading } = useGetEmailDomainsQuery( selectedSite?.ID ?? 0, {
```

Comment:
We won't fetch if selectedSite is null as provided in the queryOptions

Reply:
I still think we shouldn't default to `0`. What if we made the site id optional, and moved those query options inside the hook itself, [like this](https://github.com/Automattic/wp-calypso/blob/4d91410eb3fa0a40fb31378d9c8f72bdc51120d8/client/data/home/use-home-layout-query.ts#L16)?

Reply:
Much better idea :)